src/app/shared/components/template/components/task-progress-bar/task-progress-bar.component.ts

```diff
@@ -1,81 +1,175 @@
+  @Input() showText: boolean;
   @Output() progressStatusChange = new EventEmitter<IProgressStatus>();
   @Output() newlyCompleted = new EventEmitter<boolean>();
+  params: Partial<ITaskProgressBarParams> = {};
   subtasksTotal: number;
   subtasksCompleted: number;
-  showText = true;
+  private dataQuery$: Subscription;
 
   constructor(
     private taskService: TaskService,
     private templateFieldService: TemplateFieldService,
     private appDataService: AppDataService,
-    private campaignService: CampaignService
+    private campaignService: CampaignService,
+    private cdr: ChangeDetectorRef,
+    private dynamicDataService: DynamicDataService
   ) {
     super();
   }
 
-  ngOnInit() {
+  async ngOnInit() {
     this.getParams();
-    this.evaluateTaskGroupData();
+    if (this.params.useDynamicData) {
+      this.subscribeToData();
+    } else {
+      const taskGroupDataRows = await this.getTaskGroupDataRows();
+      this.evaluateTaskGroupData(taskGroupDataRows);
+    }
   }
 
   getParams() {
+    // If component is being explicitly instantiated from a template, get the params from the template row
     if (this._row) {
-      this.taskGroupDataList = getStringParamFromTemplateRow(this._row, "task_group_data", null);
-      this.taskGroupCompletedField = getStringParamFromTemplateRow(
+      this.params.taskGroupDataList = getStringParamFromTemplateRow(
+        this._row,
+        "task_group_data",
+        null
+      );
+      this.params.taskGroupCompletedField = getStringParamFromTemplateRow(
         this._row,
         "completed_field",
         null
       );
-      this.progressUnitsName = getStringParamFromTemplateRow(
+      this.params.progressUnitsName = getStringParamFromTemplateRow(
         this._row,
         "progress_units_name",
         "sections"
       );
+      this.params.showText = getBooleanParamFromTemplateRow(this._row, "show_text", false);
+      this.params.useDynamicData = getBooleanParamFromTemplateRow(this._row, "dynamic", false);
+      this.params.variant = getStringParamFromTemplateRow(this._row, "variant", "")
+        .split(",")
+        .join(" ") as ITaskProgressBarParams["variant"];
+    }
+    // If component is being instantiated by a parent component (e.g. task-card), use Input() values for params.
+    else {
+      this.params.taskGroupDataList = this.taskGroupDataList;
+      this.params.taskGroupCompletedField = this.taskGroupCompletedField;
+      this.params.progressUnitsName = this.progressUnitsName;
+      this.params.showText = this.showText;
     }
   }
 
   get progressPercentage() {
     return (this.subtasksCompleted / this.subtasksTotal) * 100;
   }
 
-  async evaluateTaskGroupData() {
-    const dataList = await this.appDataService.getSheet("data_list", this.taskGroupDataList);
-    const subtasks = dataList?.rows || [];
+  async getTaskGroupDataRows() {
+    const dataList = await this.appDataService.getSheet("data_list", this.params.taskGroupDataList);
```

Comment:
nit(non-blocking) - interop with other services should probably be done from the `taskService` itself and not component level, particularly so we can write tests for the behavior (although less signficant for smaller functions)